doc/source/gallery/matplotlib_scalebar.ipynb

```diff
@@ -0,0 +1,198 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import geopandas as gpd\n",
+    "from matplotlib_scalebar.scalebar import ScaleBar"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Creating a ScaleBar object"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The only required parameter for creating a ScaleBar object is *dx*. This is equal to a size of one pixel in real world. Value of this parameter depends on units of your CRS:"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Projected coordinate system (meters)\n",
+    "The easiest way to add a scale bar is using a projected coordinate system with meters as units. Just set `dx = 1`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "nybb = gpd.read_file(gpd.datasets.get_path('nybb'))\n",
+    "nybb.to_crs(32619, inplace=True)  # Must be converted to a projected coordinate system\n",
+    "\n",
+    "ax = nybb.plot()\n",
+    "ax.add_artist(ScaleBar(1))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Geographic coordinate system (degrees)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "With a geographic coordinate system with degrees as units, *dx* should be equal to a distance in meters of two points with the same latitude (Y coordinate) which are one full degree of longitude (X) apart. You can calculate this distance with an online calculator [(e.g. Great Circle calculator)](http://edwilliams.org/gccalc.htm) or in geopandas:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from shapely.geometry.point import Point\n",
+    "\n",
+    "nybb = gpd.read_file(gpd.datasets.get_path('nybb'))\n",
+    "nybb.to_crs(4326, inplace=True)  # Using WGS 84 (in degrees)\n",
+    "\n",
+    "# Creating the points and calculating the distance\n",
+    "points = gpd.GeoSeries([Point(-73.5, 40.5), Point(-74.5, 40.5)], crs=4326)\n",
+    "# Points are on the same latitude, one degree of longitude apart, with roughly the coordinates of NYC\n",
+    "points = points.to_crs(32619)  # Points are converted from WGS 84 to a projected system for\n",
+    "# the calculation\n",
+    "\n",
+    "distance_meters = points[0].distance(points[1])\n",
+    "# The result slightly differs from Great Circle calculator (84,921 and 84,767 meters)\n",
+    "# but the difference is insignificant\n",
+    "\n",
+    "ax = nybb.plot()  # The final plot uses WGS 84 again\n",
+    "ax.add_artist(ScaleBar(distance_meters))"
+   ]
+  },
```

Comment:
Can you split this cell into multiple and have those comments as markdowns cells between code cells?